javascripts/2048.js

```diff
@@ -1,32 +1,383 @@
+    var x = Math.floor(Math.random() * (boardSize));
+    if(board[y][x] == 0) {
+      newTile = [y,x];
+      var rando = Math.random();
+      if(rando < .9) {
+        board[y][x] = 2;
+      }
+      else{
+        board[y][x] = 4;
+      }
+    }
+    else {
+      assignRandoTiles();
+      // this will run endlessly when the board is full -- write a conditional for that
+    }
+  }
+
+  function displayBoard() {
+    for(var y = 0; y < boardSize; y++) {
+      for(var x = 0; x < boardSize; x++) {
+        var cell = $(".tile[data-row=y" + y + "][data-col=x" + x);
+        cell.attr("data-val", board[y][x]);
+        if(board[y][x] == 0){
+          cell.text("");
+        }
+        else {
+          cell.text(board[y][x]);
+        }
+        if(y == newTile[0] && x == newTile[1]){
+          cell.addClass("popper");
+          cell.one("animationend", function(e) {
+            $(this).removeClass("popper");
+          });
+        }
+      }
+    }
+    newTile = [];
+  }
+
+
+function moveTile(direction) {
   switch(direction) {
     case 38: //up
-      tile.attr("data-row","r0");
+      moveUp();
       break;
     case 40: //down
-      tile.attr("data-row","r3");
+      moveDown();
       break;
     case 37: //left
-      tile.attr("data-col","c0");
+      moveLeft();
       break;
     case 39: //right
-      tile.attr("data-col","c3");
+      moveRight();
       break;
   }
+  displayBoard();
+}
+
+function moveUp() {
+  var starterBoard = board.toString();
+  setupTempBoard("up");
+  // traversal starting point = y3, x0
+  for(y = 3; y >= 0; y--) {
+    for(x = 0; x < boardSize; x++) {
+      var next = checkNext(y, x, "up");
+      if(board[y][x] == 0 && next != null) {
+        board[y][x]     = board[y - 1][x];
+        board[y - 1][x] = 0;
+      }
+      else if(next == board[y][x]) {
+        board[y][x]    += board[y - 1][x]; // collapse it
+        score += board[y][x];
+        // populateScore(score);
+        board[y - 1][x] = 0;
+      }
+    }
+  }
+  var endGame = checkEndGame();
+  displayGameResult(endGame);
+
+  if (board.toString() != starterBoard) {
+    assignRandoTiles();
+  }
+}
+
+function moveDown () {
+  starterBoard = board.toString();
+  setupTempBoard("down");
+  // traversal starting point = y0, x0
+  for(y = 0; y < boardSize; y++) {
+    for(x = 0; x < boardSize; x++) {
+      var next = checkNext(y, x, "down");
+      if(board[y][x] == 0 && next != null) {
+        board[y][x]     = board[y + 1][x];
+        board[y + 1][x] = 0;
+      }
+      else if(next == board[y][x]) {
+        board[y][x]    += board[y + 1][x]; // collapse it
+        score += board[y][x];
+        board[y + 1][x] = 0;
+      }
+    }
+  }
+  var endGame = checkEndGame();
+  displayGameResult(endGame);
+
+  if (board.toString() != starterBoard) {
+    assignRandoTiles();
+  }
+}
+
+function moveLeft() {
+  starterBoard = board.toString();
+  setupTempBoard("left");
+  // traversal starting point = y0, x3
```

Comment:
This comment appears to be copied and pasted, and is not accurate.

Misleading comments are more dangerous than no comments at all.

Be sure to update your comments when you copy code.